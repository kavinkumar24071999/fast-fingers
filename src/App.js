import './App.css';
import InputBar from './components/inputBar/InputBar';
import Timer from './components/timer/Timer';
import WordsBar from "./components/wordsBar/WordsBar";
import {useEffect, useState} from "react";
import words from "./data/data";

export default App;

const EMPTY_WORD = '';
const SPACE = " ";
const MAX_WORDS = 10;

function App() {
  const [isTimerRunning, setTimerRunning] = useState(false);
  const [inputWord, setInputWord] = useState(EMPTY_WORD);
  const [currentWordIndex, setCurrentWordIndex] = useState(0);
  const [wordPermanentStyle, setWordPermanentStyle] = useState({});
  const getRandomWords = () => {
    return words.sort(() => (Math.random() > .5) ? 1 : -1);
  };

  const [randomWords, setRandomWords] = useState([]);

  useEffect(() => {
    isTimerRunning && setRandomWords(getRandomWords().slice(0, MAX_WORDS));
  }, [isTimerRunning]);

  useEffect(() => {
    if (inputWord.includes(SPACE)) {
      wordPermanentStyleHandler();
      setInputWord(EMPTY_WORD);
      incrementWordIndexByOne();
    }
    if (currentWordIndex >= MAX_WORDS) {
      setRandomWords(getRandomWords().slice(0, MAX_WORDS));
      setCurrentWordIndex(0);
      setWordPermanentStyle({});
    }
  }, [inputWord]);

  const incrementWordIndexByOne = () => {
    setCurrentWordIndex(currentWordIndex + 1);
  };

  const onTimerStart = () => {
    setTimerRunning(true);
  };

  const wordPermanentStyleHandler = () => {
    if (randomWords[currentWordIndex].length === inputWord.trim().length
      && randomWords[currentWordIndex] === inputWord.trim()) {
      setWordPermanentStyle(wordPermanentStyle => {
        wordPermanentStyle[currentWordIndex] = 'correct';
        return wordPermanentStyle;
      });
    } else {
      setWordPermanentStyle(wordPermanentStyle => {
        wordPermanentStyle[currentWordIndex] = 'incorrect';
        return wordPermanentStyle;
      });
    }
  };

  const onTimerEnd = () => {
    setCurrentWordIndex(0);
    setWordPermanentStyle({});
    setTimerRunning(false);
  };

  return (
    <div className="App">
      {isTimerRunning && <WordsBar inputWord={inputWord} wordIndex={currentWordIndex} randomWords={randomWords}
        wordPermanentStyle={wordPermanentStyle} />}
      <InputBar word={inputWord} setWord={setInputWord} />
      <Timer maxSeconds={60} onTimerStart={onTimerStart} onTimerEnd={onTimerEnd} />
    </div>
  );
}

