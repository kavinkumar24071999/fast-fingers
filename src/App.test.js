import React from 'react';
import Enzyme, {shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

Enzyme.configure({adapter: new Adapter()});

describe('App', function () {
  it('should have home page', function () {
    const wrapper = shallow(<App />);
    expect(wrapper.find('header').text()).toEqual('Learn React');
    expect(wrapper.html()).toMatchSnapshot();
  });
});