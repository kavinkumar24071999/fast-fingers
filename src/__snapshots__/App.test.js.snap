// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `"<div class=\\"App\\"><div class=\\"input-bar\\"><input value=\\"\\"/></div><div class=\\"Timer\\"><button>Start</button><div class=\\"seconds\\">Remaining Seconds: 60</div></div></div>"`;
